packages/transaction-controller/src/TransactionController.ts

```diff
@@ -950,6 +973,24 @@ export class TransactionController extends BaseController<
         transactionMeta,
         'TransactionController:confirmExternalTransaction - Add external transaction',
       );
+
+      if (transactionMeta.type === TransactionType.swap) {
+        updatePostTransactionBalance(transactionMeta, {
+          ethQuery: this.ethQuery,
+          getTransaction: this.getTransaction.bind(this),
+          updateTransaction: this.updateTransaction.bind(this),
+        })
+          .then(({ updatedTransactionMeta, approvalTransactionMeta }) => {
+            this.hub.emit(TransactionEvent.postTransactionBalanceUpdated, {
+              transactionMeta: updatedTransactionMeta,
+              approvalTransactionMeta,
+            });
+          })
+          .catch((error) => {
+            /* istanbul ignore next */
+            console.error(error);
```

Comment:
Shall we use the `log` object instead so we don't surface these in production as errors here seem unlikely and low priority?

Should we also add a message such as `Error while updating post transaction balance` ?

Reply:
Done https://github.com/MetaMask/core/pull/1877/commits/62aed8af2659145cd801befe8578593f47a0cba5